controller/test/blower_fsm/blower_fsm_test.cpp

```diff
@@ -88,142 +110,157 @@ TEST(BlowerFsmTest, PressureControl) {
+      {
+          {0, 10},    // start at PEEP
+          {500, 20},  // rise to PIP
+          {1000, 20}, // end inspire at PIP
+          {1100, 10}, // fall to PEEP
+          {1500, 10},
+      });
 
-  // - when flow is zero: breath is triggered on expire_deadline_ rather than
-  // patient triggered, to enforce minimum respiratory rate
-  // - when flow is breath: trigger breath if in expire mode
-  testSequence({
-      // first breath is mandatory, starts out at PEEP and rises to PIP over
-      // period of RISE_TIME
-      {p, inputs_zero, 0, cmH2O(10), FlowDirection::INSPIRATORY},
-      {p, inputs_zero, rise_time_ms / 4, cmH2O(12.5),
-       FlowDirection::INSPIRATORY},
-      {p, inputs_zero, rise_time_ms / 2, cmH2O(15), FlowDirection::INSPIRATORY},
-      {p, inputs_zero, 3 * rise_time_ms / 4, cmH2O(17.5),
-       FlowDirection::INSPIRATORY},
-      {p, inputs_zero, rise_time_ms, cmH2O(20), FlowDirection::INSPIRATORY},
-      // breath has no effect during inspire phase
-      {p, inputs_breath, 1000, cmH2O(20), FlowDirection::INSPIRATORY},
-      {p, inputs_zero, 1999, cmH2O(20), FlowDirection::INSPIRATORY},
-      {p, inputs_zero, 2001, cmH2O(10), FlowDirection::EXPIRATORY},
-      // need to run with non-breath flow while already in exhale leg to
-      // initialize detection threshold
-      {p, inputs_zero, 2002, cmH2O(10), FlowDirection::EXPIRATORY},
-      // check that calling with zero flow before the end of the breath does not
-      // tigger the next breath
-      {p, inputs_zero, 2500, cmH2O(10), FlowDirection::EXPIRATORY},
-      {p, inputs_zero, 2999, cmH2O(10), FlowDirection::EXPIRATORY},
-      // trigger breath on expire_deadline_ (wise rise over RISE_TIME)
-      {p, inputs_zero, 3001, cmH2O(10), FlowDirection::INSPIRATORY},
-      {p, inputs_breath, 3001 + rise_time_ms / 2, cmH2O(15),
-       FlowDirection::INSPIRATORY},
-      {p, inputs_breath, 3001 + rise_time_ms, cmH2O(20),
-       FlowDirection::INSPIRATORY},
-      {p, inputs_breath, 4999, cmH2O(20), FlowDirection::INSPIRATORY},
-      {p, inputs_zero, 5001, cmH2O(10), FlowDirection::EXPIRATORY},
-      // need to run with non-breath flow while already in exhale leg to
-      // initialize detection threshold
-      {p, inputs_zero, 5002, cmH2O(10), FlowDirection::EXPIRATORY},
-      // triggered breath
-      {p, inputs_breath, 5200, cmH2O(10), FlowDirection::INSPIRATORY},
-      {p, inputs_breath, 5200 + rise_time_ms / 2, cmH2O(15),
-       FlowDirection::INSPIRATORY},
-      {p, inputs_breath, 5200 + rise_time_ms, cmH2O(20),
-       FlowDirection::INSPIRATORY},
-      // triggered breath is of normal duration
-      {p, inputs_zero, 7199, cmH2O(20), FlowDirection::INSPIRATORY},
-      {p, inputs_zero, 7201, cmH2O(10), FlowDirection::EXPIRATORY},
-  });
+  ASSERT_TRUE(results.expire_start_time.has_value());
+  ASSERT_TRUE(results.finish_time.has_value());
+  EXPECT_EQ(results.expire_start_time->milliseconds(), 1000.f);
```

Comment:
Expecting exact equality seems fishy, is this dependent on the algorithm or only on the setpoint checks?

Reply:
In pressure control the machine controls the duration of inspiration and expiration precisely.  In pressure assist, the patient controls the expire duration, but the machine still controls the inspire duration.

So I think this is basically correct -- if I understand what you're questioning.